pkg/repository.go

```diff
@@ -0,0 +1,76 @@
+package pkg
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/go-redis/redis/v8"
+	log "github.com/sirupsen/logrus"
+)
+
+//struct which represent mainly interaction with redis
+type RedisRepository struct {
+	Client *redis.Client
+	Ctx  context.Context
+}
+
+
+//should update existing history with order or create a new one 
+func (r *RedisRepository) SaveOrder(userId string, order Order) error {
+	val, err := r.Client.Get(r.Ctx, (userId)).Result()
+	if err != nil {
+		//create new user with user history 
+		history := OrderHistory{Orders: []Order{order}}
+		r.saveHistory(userId,&history)
+	}else {
+		// update existing user history 
+		var history OrderHistory
+		json.Unmarshal([]byte(val), &history)
+		history.add(order)
+		r.saveHistory(userId, &history)	
+	}
+	return nil		
+}
+
+//should add new history to repository (ACHTUNG it will replace the old one)
+//and also it will clean orders which are older than 24 hours 
+func (r *RedisRepository) saveHistory(userId string,history *OrderHistory) error {
+	newHistory := CleanWithinWindow(history, 24)
+	
+	data, jsonError := json.Marshal(newHistory)
+	if jsonError != nil {
+		log.Error(fmt.Sprintf("Erorr while saving history for user with [%s]",userId))
+		log.Error(jsonError)
+		return jsonError
+	}
+	err2 := r.Client.Set(r.Ctx, userId, string(data),time.Duration(24*time.Hour)).Err()
```

Comment:
TTL is hardcoded. 24 * time.Hour.
Is it ok? our quotas can be based on different interval, not only 24 * time.Hour

Reply:
This property should be in a configuration, my bad, but yes if interval of the quota is higher than 24hrs than this field also should be increased